modules/navier_stokes/test/src/postprocessors/TestFaceToCellReconstruction.C

```diff
@@ -0,0 +1,71 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "TestFaceToCellReconstruction.h"
+
+// MOOSE includes
+#include "MooseMesh.h"
+
+registerMooseObject("NavierStokesTestApp", TestFaceToCellReconstruction);
+
+InputParameters
+TestFaceToCellReconstruction::validParams()
+{
+  InputParameters params = GeneralPostprocessor::validParams();
+  params.addClassDescription("Computes reconstruction error for face centered functor.");
+  return params;
+}
+
+TestFaceToCellReconstruction::TestFaceToCellReconstruction(const InputParameters & parameters)
+  : GeneralPostprocessor(parameters),
+    _reconstruction_error(0.0),
+    _face_values(_subproblem.mesh(), _subproblem.mesh().meshSubdomains(), "face_values")
+{
+}
+
+void
+TestFaceToCellReconstruction::initialize()
+{
+  for (auto & fi : _fe_problem.mesh().faceInfo())
```

Comment:
So I was looking through this PR wondering at the motivation for the change in what `faceInfo()` holds and the addition of the variational iterators, and I'm thinking that this is it? To avoid doing a communication after filling the truly local face elements? This is fine in this use case, but I'm concerned that users of this functor may fill a non-local map entry on one process which is local to another process which fills in a different value. This could be addressed with some parallel consistency checks, but maybe I'm being overly paranoid.

 The other consideration is: what is more expensive? Map entry computation or communication? Probably in most cases it will be the latter which would suggest your design is a good one ... but then you have to make a trade with the cost of filtering through non-local face informations when computing your owned-ranges.

Anyway this is a long monologue ... I think what you've done is probably perfectly fine. If something related to this shows up in profiling we can revert the framework level changes and do communication when building the face map functors